interfaces/builtin/ofono.go

```diff
@@ -0,0 +1,329 @@
+/dev/tty[^0-9]* rw,
+/dev/cdc-* rw,
+/dev/modem* rw,
+/dev/dsp rw,
+/dev/chnlat11 rw,
+/dev/socket/rild* rw,
+
+network netlink raw,
+network netlink dgram,
+network bridge,
+network inet,
+network inet6,
+network packet,
+network bluetooth,
+
+include <abstractions/nameservice>
+
+# DBus accesses
+include <abstractions/dbus-strict>
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member={Request,Release}Name
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.ofono",
+
+# Allow traffic to/from our path and interface with any method for unconfined
+# clients to talk to our ofono services.
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    interface=org.ofono.*
+    peer=(label=unconfined),
+`
+
+const ofonoConnectedSlotAppArmor = `
+# Allow service to interact with connected clients
+
+# Allow traffic to/from our interfaces. The path depends on the modem plugin,
+# and is arbitrary.
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    interface=org.ofono.*
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`
+
+const ofonoConnectedPlugAppArmor = `
+# Description: Allow using Ofono service. Reserved because this gives
+#  privileged access to the Ofono service.
+# Usage: reserved
+
+#include <abstractions/dbus-strict>
+
+# Allow all access to ofono services
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    interface=org.ofono.*
+    peer=(label=###SLOT_SECURITY_TAGS###),
+`
+
+const ofonoConnectedPlugAppArmorClassic = `
+# Allow access to the unconfined ofono services on classic.
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    interface=org.ofono.*
+    peer=(label=unconfined),
+`
+
+const ofonoPermanentSlotSecComp = `
+# Description: Allow operating as the ofono service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# Communicate with DBus, netlink, rild
+accept
+accept4
+bind
+getsockopt
+listen
+recv
+recvfrom
+recvmmsg
+recvmsg
+send
+sendmmsg
+sendmsg
+sendto
+shutdown
+`
+
+const ofonoConnectedPlugSecComp = `
+# Description: Allow using ofono service. Reserved because this gives
+#  privileged access to the ofono service.
+# Usage: reserved
+
+# Can communicate with DBus system service
+recv
+recvmsg
+recvfrom
+send
+sendto
+sendmsg
+`
+
+const ofonoPermanentSlotDBus = `
+<!-- Comes from src/ofono.conf in sources -->
+
+<policy user="root">
+  <allow own="org.ofono"/>
+  <allow send_destination="org.ofono"/>
+  <allow send_interface="org.ofono.SimToolkitAgent"/>
+  <allow send_interface="org.ofono.PushNotificationAgent"/>
+  <allow send_interface="org.ofono.SmartMessagingAgent"/>
+  <allow send_interface="org.ofono.PositioningRequestAgent"/>
+  <allow send_interface="org.ofono.HandsfreeAudioAgent"/>
+</policy>
+
+<policy at_console="true">
+  <allow send_destination="org.ofono"/>
+</policy>
```

Comment:
It looks like 'at_console' is deprecated: https://lists.fedoraproject.org/pipermail/devel/2014-May/199004.html. Considering the lack of policykit on Ubuntu Core it seems like removing it and relying on the root and default policy would be wise. Can you comment on why 'at_console' is here?